tests/unit-ext/test-ext-arg.c

```diff
@@ -211,6 +215,44 @@ test_validator2_handler (const jerry_value_t func_obj_val __attribute__((unused)
   return jerry_create_undefined ();
 } /* test_validator2_handler */
 
+/**
+ * Convert all props of an JS object into native variables.
+ */
+static jerry_value_t
+test_validator_prop_handler (const jerry_value_t func_obj_val __attribute__((unused)), /**< function object */
+                             const jerry_value_t this_val __attribute__((unused)), /**< this value */
+                             const jerry_value_t args_p[], /**< arguments list */
+                             const jerry_length_t args_cnt __attribute__((unused))) /**< arguments length */
+{
+  bool native1 = false;
+  double native2 = 0;
+  double native3 = 3;
+
+  const char *name_p[] = {"prop1", "prop2", "prop3"};
+
+  jerryx_arg_t mapping[] =
+  {
+    jerryx_arg_boolean (&native1, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    jerryx_arg_number (&native2, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    jerryx_arg_number (&native3, JERRYX_ARG_COERCE, JERRYX_ARG_OPTIONAL)
+  };
+
+  jerry_value_t is_ok = jerryx_arg_transform_object_properties (args_p[0],
```

Comment:
My example:
in binding.c, the function handler which use color
```c
typedef struct
{
uint8 red;
uint8 green,
uint8 blue;
uint8 alpha
} native_color_t;

jerry_value_t
set_color_handler(
	const jerry_value_t	func_obj_val,
	const jerry_value_t	this_val,
	const jerry_value_t	args[],
	const jerry_length_t	 args_cnt)
{
	native_view_t *view_p;
	native_color_t color;

	jerryx_arg_t mapping[] =
	{
		jerryx_arg_native_pointer(
			(void **)&view_p,
			&JERRY_NATIVE_HANDLE_INFO_FOR_CTYPE(view),
			JERRYX_ARG_REQUIRED
			),
		jerryx_arg_custom(&color, 0, js_arg_check_is_color)
	};

	jerry_value_t is_ok = jerryx_arg_transform_args(args, args_cnt, mapping,
							2);
      ....
```

and the js_arg_check_is_color is my custom function
```c
jerry_value_t
js_arg_check_is_color(jerryx_arg_js_iterator_t *js_arg_iter_p,
		const jerryx_arg_t *	c_arg_p)
{
	jerry_value_t js_arg = jerryx_arg_js_iterator_pop (js_arg_iter_p);
	const char *keys[] = {"r", "g", "b", "a"};
	native_color_t *color_p = c_arg_p->dest;

	const jerryx_arg_t mapping[] = {
	jerryx_arg_uint8 (&color_p->red, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
	jerryx_arg_uint8 (&color_p->green, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
	jerryx_arg_uint8 (&color_p->blue, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
	jerryx_arg_uint8 (&color_p->alpha, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
	};
	
	return jerryx_arg_transform_object_properties(js_arg, keys, 4, mapping, 4);
}
```